core/client/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -236,7 +235,5 @@ private BlockInStream(PacketInStream inputStream, BlockWorkerClient blockWorkerC
 
     mCloser = closer;
     mCloser.register(mInputStream);
-    mLocal = blockWorkerClient.getDataServerAddress().getHostName()
-        .equals(NetworkAddressUtils.getClientHostName());
```

Comment:
This worked before, but the source of truth for whether the `BlockInStream` is local should be the stream it's based on, not logic in `BlockInStream`. If the logic for determining locality changed in the underlying stream, this could easily turn into a bug.